### PR TITLE
Cancel analysis by name

### PR DESCRIPTION
**Pull request by patrikgrenfeldt**
This PR adds the feature to cancel analysis by name

**How to test**:
1. install on stage hasta
1. activate stage: `us`
1. show all running analyses
1. run following command: `trailblazer cancel -j -c [case-id]`
1. run following command: `trailblazer cancel [run-id] -j`

**Expected outcome**:
`trailblazer cancel -j -c [case-id]` should print to screen the jobs that would have been cancelled. The flag -c should remove the same jobs as if you gave it the integer run-id. `trailblazer cancel [run-id] -j` should print to screen the same jobs.
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [x] tests executed by @b4ckm4n 
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

This is minor **version bump** because we add a feature in a backwards compatible manner

### CODE REVIEW

**Comment by b4ckm4n** [comment]
When executing How to test:

$ trailblazer ls -s running
157399 | hardyosprey 2020-01-27 [WGS/RUNNING] - 21.4286/100
157400 | quickmolly 2020-01-27 [WGS/RUNNING] - 4.41176/100
147610 | purerodent 2019-10-31 [/RUNNING] - 3.125/100

$ trailblazer cancel -j -c quickmolly
327332
327333
327334
327335
327336
327337
327338
327339
327340
327341
327342
327343
327344
327345
327346
327347
327348
327349
327350
327351
327352
327353
327354
327355
327356
327357
327358
327359
327360
327361
327362
327363
327364
327365
327366
327367
327368
327369
327370
327371
327372
327373
327374
327375
327376
327377
327378
327379
327380
327381
327382
327383
327384
327385
327386
327387
327388
327389
327390
327391
327392
327393
327394
327395

$ trailblazer cancel 157400 -j
327332
327333
327334
327335
327336
327337
327338
327339
327340
327341
327342
327343
327344
327345
327346
327347
327348
327349
327350
327351
327352
327353
327354
327355
327356
327357
327358
327359
327360
327361
327362
327363
327364
327365
327366
327367
327368
327369
327370
327371
327372
327373
327374
327375
327376
327377
327378
327379
327380
327381
327382
327383
327384
327385
327386
327387
327388
327389
327390
327391
327392
327393
327394
327395

**Comment by henrikstranneheim** [comment]
@patrikgrenfeldt This looks great. We need to start driving the unit test up for TB. Could you add a unit test on the CLI command?

**Comment by henrikstranneheim** [comment]
Redid the test: 🏁 
Screenshot 2020-08-25 at 09 30 58

```
trailblazer cancel -j -c coolsole
561681
561682
561683
561684
561685
561686
561687
561688
561689
561690
561691
561692
561693
561694
561695
561696
561697
561698
561699
561700
561701
561702
561703
561704
561705
561706
561707
561708
561709
561710
561711
561712
561713
561714
561715
561716
561717
561718
561719
561720
561721
561722
561723
561724
561725
561726
561727
561728
561729
561730
561731
561732
561733
561734
561735
561736
561737
561738
561739
561740
561741
561742
561743
561744
561745
561746
561747
561748
561749
561750
561751
561752
561753
561754
561755
561756
561757
561758
561759
561760
561761
561762
561763
561764
561765
561766
561767
561768
561769
561770
561771
561772
561773
561774
561775
561776
561777
561778
561779
561780
561781
561782
561783
561784
561785
561786
561787
561788
561789
561790
561791
561792
561793
561794
561795
561796
[3|109|638] [henrik.stranneheim@hasta:~] [S_main] 11s $ trailblazer cancel -j 220708
561681
561682
561683
561684
561685
561686
561687
561688
561689
561690
561691
561692
561693
561694
561695
561696
561697
561698
561699
561700
561701
561702
561703
561704
561705
561706
561707
561708
561709
561710
561711
561712
561713
561714
561715
561716
561717
561718
561719
561720
561721
561722
561723
561724
561725
561726
561727
561728
561729
561730
561731
561732
561733
561734
561735
561736
561737
561738
561739
561740
561741
561742
561743
561744
561745
561746
561747
561748
561749
561750
561751
561752
561753
561754
561755
561756
561757
561758
561759
561760
561761
561762
561763
561764
561765
561766
561767
561768
561769
561770
561771
561772
561773
561774
561775
561776
561777
561778
561779
561780
561781
561782
561783
561784
561785
561786
561787
561788
561789
561790
561791
561792
561793
561794
561795
561796
```

File: trailblazer/cli/core.py [hunk 1]
```diff
@@ -160,12 +160,23 @@ def user(context, name, email):
...
-@click.option("-j", "--jobs", is_flag=True, help="only print job ids")
-@click.argument("analysis_id", type=int)
+@click.option('-j', '--jobs', is_flag=True, help='only print job ids')
+@click.argument('analysis_id', type=int, required=False)
+@click.option('-c', '--case', 'case_id', type=str, help="cancel latest running analysis of case")
 @click.pass_context
-def cancel(context, jobs, analysis_id):
+def cancel(context, jobs, analysis_id=None, case_id=None):
+
     """Cancel all jobs in a run."""
```
Comment (henrikstranneheim) [inline trailblazer/cli/core.py:169]:
```suggestion
    """Cancel all jobs in a run"""
```

File: trailblazer/cli/core.py [hunk 2]
```diff
@@ -160,12 +160,23 @@ def user(context, name, email):
...
-    analysis_obj = context.obj["store"].analysis(analysis_id)
+    if analysis_id:
+        analysis_obj = context.obj['store'].analysis(analysis_id)
+    elif case_id:
+        analyses = context.obj['store'].analyses(family=case_id, status="running")
+        if not analyses or not len(analyses.all()) == 1:
+            click.echo(click.style(f'{len(analyses.all())} running analyses found, 1 expected',
+                                   fg='yellow'))
+            context.abort()
+        analysis_obj = analyses.first()
```
Comment (henrikstranneheim) [inline trailblazer/cli/core.py:178]: I think it would be great if we put this in a function, to enable unit testing

**Review by henrikstranneheim — COMMENTED** [review]
This would be a nice addition indeed - well done! We need to start writing more tests for trailblazer. It would be great with some unit test over this new functionality and a CLI test for the click command. What do you think?

**Comment by henrikstranneheim** [comment]
@moonso Why close this one?

**Comment by moonso** [comment]
Ah sorry, should have given a comment. We took the decision on the operations meeting today. Since trailblazer has been more or less rewritten this code will not work. @Mropat and @elevu will implement this feature in another PR

**Comment by patrikgrenfeldt** [comment]
This code still works! Reopening :)

**Comment by moonso** [comment]
Well I don't mind, thought @patrikgrenfeldt and @Mropat agreed to close yesterday

**Comment by patrikgrenfeldt** [comment]
Great, would you review the @moonso ?

**Comment by moonso** [comment]
> Great, would you review the @moonso ?

I think it's better if @Mropat does that since she is really into the code

**Comment by Mropat** [comment]
There is a new implementation for this. The prompt and the ongoing jobs will be visible in the web interface. There is a command for CLI cancelling analysis by analysis ID

**Pull request closed without merging**